### PR TITLE
EAP7-1780 - imagestream for JDK 17

### PR DESCRIPTION
**Pull request by luck3y**
https://issues.redhat.com/browse/EAP7-1780
Signed-off-by: Ken Wills <kwills@redhat.com>

### CODE REVIEW

File: eap74-openjdk17-image-stream.json [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    "items": [
+        {
+            "kind": "ImageStream",
+            "apiVersion": "image.openshift.io/v1",
+            "metadata": {
+                "name": "jboss-eap74-openjdk17-openshift",
+                "annotations": {
+                    "openshift.io/display-name": "JBoss EAP 7.4.0 with OpenJDK 17",
+                    "openshift.io/provider-display-name": "Red Hat, Inc.",
+                    "version": "7.4.0"
```
Comment (jmesnil) [inline eap74-openjdk17-image-stream.json]: version looks outdated, let's use `7.4.7` based on https://catalog.redhat.com/software/containers/jboss-eap-7/eap74-openjdk17-openshift-rhel8/62eb82191b5d152aa19206ef?tag=7.4.7-4&push_date=1664988439000

File: eap74-openjdk17-image-stream.json [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                }
+            },
+            "labels": {
+                "xpaas": "7.4.0"
```
Comment (jmesnil) [inline eap74-openjdk17-image-stream.json]: idem

File: eap74-openjdk17-image-stream.json [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                        "referencePolicy": {
+                            "type": "Local"
+                        },
+                        "from": {
+                            "kind": "DockerImage",
+                            "name": "registry.redhat.io/jboss-eap-7/eap74-openjdk17-openshift-rhel8:latest"
+                        }
+                    },
+                    {
+                        "name": "7.4.0",
```
Comment (jmesnil) [inline eap74-openjdk17-image-stream.json]: idem

File: eap74-openjdk17-image-stream.json [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                        "annotations": {
+                            "description": "The latest available build of JBoss EAP 7.4.0 with OpenJDK 17.",
```
Comment (jmesnil) [inline eap74-openjdk17-image-stream.json]: idem

File: eap74-openjdk17-image-stream.json [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+                            "iconClass": "icon-eap",
+                            "tags": "builder,eap,javaee,java,jboss,hidden",
+                            "supports": "eap:7.4.0,javaee:8,java:17",
```
Comment (jmesnil) [inline eap74-openjdk17-image-stream.json]: idem

File: eap74-openjdk17-image-stream.json [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+                            "sampleRepo": "https://github.com/jbossas/eap-quickstarts/",
+                            "sampleContextDir": "kitchensink",
+                            "sampleRef": "7.4.x",
+                            "version": "latest",
+                            "openshift.io/display-name": "JBoss EAP 7.4.0 with OpenJDK17"
```
Comment (jmesnil) [inline eap74-openjdk17-image-stream.json]: idem

File: eap74-openjdk17-image-stream.json [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+                        },
+                        "referencePolicy": {
+                            "type": "Local"
+                        },
+                        "from": {
+                            "kind": "DockerImage",
+                            "name": "registry.redhat.io/jboss-eap-7/eap74-openjdk17-openshift-rhel8"
```
Comment (jmesnil) [inline eap74-openjdk17-image-stream.json]: do you have to use `registry.redhat.io/jboss-eap-7/eap74-openjdk17-openshift-rhel8:7.4.7` here?

File: eap74-openjdk17-image-stream.json [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+                ]
+            }
+        },
+        {
+            "kind": "ImageStream",
+            "apiVersion": "image.openshift.io/v1",
+            "metadata": {
+                "name": "jboss-eap74-openjdk17-runtime-openshift",
+                "annotations": {
+                    "openshift.io/display-name": "JBoss EAP 7.4.0 Runtime Image with OpenJDK 17",
```
Comment (jmesnil) [inline eap74-openjdk17-image-stream.json]: idem

File: eap74-openjdk17-image-stream.json [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+                    "openshift.io/provider-display-name": "Red Hat, Inc.",
+                    "version": "7.4.0"
```
Comment (jmesnil) [inline eap74-openjdk17-image-stream.json]: idem

File: eap74-openjdk17-image-stream.json [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+                }
+            },
+            "labels": {
+                "xpaas": "7.4.0"
```
Comment (jmesnil) [inline eap74-openjdk17-image-stream.json]: idem

File: eap74-openjdk17-image-stream.json [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+                        "referencePolicy": {
+                            "type": "Local"
+                        },
+                        "from": {
+                            "kind": "DockerImage",
+                            "name": "registry.redhat.io/jboss-eap-7/eap74-openjdk17-runtime-openshift-rhel8:latest"
+                        }
+                    },
+                    {
+                        "name": "7.4.0",
```
Comment (jmesnil) [inline eap74-openjdk17-image-stream.json]: idem

File: eap74-openjdk17-image-stream.json [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+                            "sampleRef": "7.4.x",
+                            "version": "latest",
+                            "openshift.io/display-name": "JBoss EAP 7.4.0 with OpenJDK 17 (Runtime Image)"
+                        },
+                        "referencePolicy": {
+                            "type": "Local"
+                        },
+                        "from": {
+                            "kind": "DockerImage",
+                            "name": "registry.redhat.io/jboss-eap-7/eap74-openjdk17-runtime-openshift-rhel8:latest"
```
Comment (jmesnil) [inline eap74-openjdk17-image-stream.json:128]: idem

**Review by jmesnil — CHANGES_REQUESTED** [review]
I think we need to update the image stream to start the tag with `7.4.7`.

File: eap74-openjdk17-image-stream.json (hunk 1 above)
Comment (luck3y) [reply eap74-openjdk17-image-stream.json]: as discussed, using :7.4 instead

**Review by jmesnil — APPROVED** [review]

**Pull request merged**